jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2585,34 +2585,38 @@ void getBytes(int start, int end, char[] data, int index) {
 	 */
 	public void getChars(int start, int end, char[] data, int index) {
 		if (0 <= start && start <= end && end <= lengthInternal()) {
-			// Check if the String is compressed
-			if (enableCompression && (null == compressionFlag || count >= 0)) {
-				decompress(value, start, data, index, end - start);
-			} else {
-				/*[IF Sidecar19-SE]*/
-				decompressedArrayCopy(value, start, data, index, end - start);
-				/*[ELSE]*/
-				System.arraycopy(value, start, data, index, end - start);
-				/*[ENDIF]*/
-			}
+			getCharsImpl(start, end, data, index);
 		} else {
 			throw new StringIndexOutOfBoundsException();
 		}
 	}
 
+	void getCharsImpl(int start, int end, char[] data, int index) {
+		// Check if the String is compressed
+		if (enableCompression && (null == compressionFlag || count >= 0)) {
+			decompress(value, start, data, index, end - start);
+		} else {
+			decompressedArrayCopy(value, start, data, index, end - start);
+		}
+	}
+
 	/*[IF Sidecar19-SE]*/
 	void getChars(int start, int end, byte[] data, int index) {
 		if (0 <= start && start <= end && end <= lengthInternal()) {
-			// Check if the String is compressed
-			if (enableCompression && (null == compressionFlag || count >= 0)) {
-				decompress(value, start, data, index, end - start);
-			} else {
-				decompressedArrayCopy(value, start, data, index, end - start);
-			}
+			getCharsImpl(start, end, data, index);
 		} else {
 			throw new StringIndexOutOfBoundsException();
 		}
 	}
+
+	void getCharsImpl(int start, int end, byte[] data, int index) {
```

Comment:
This implementation and the one at line 2594 differ only in the signature.  Can you use the preprocessor to have a single method body with the signature conditionally controlled?

Reply:
For Java 9, both byte[] and char[] version of getCharsImpl need to be present, as `public void getChars(int start, int end, char[] data, int index)` is a public method which will have to call the char version of getCharsImpl.